src/Device/Port/TTYEnumerator.cpp

```diff
@@ -24,34 +24,49 @@ Copyright_License {
 #include "TTYEnumerator.hpp"
 #include "util/CharUtil.hxx"
 #include "util/StringCompare.hxx"
+#include "LogFile.hpp"
 
 #include <stdio.h>
 #include <unistd.h>
+#include <sys/stat.h>
+#include <tchar.h>
 
-[[gnu::pure]]
 static bool
-CheckTTYName(const char *name) noexcept
+IsDirectoryPresent(const char *path)
 {
-  /* filter "/dev/tty*" */
-  if (const char *t = StringAfterPrefix(name, "tty"); t != nullptr) {
-    if (*t == 0)
-      /* ignore /dev/tty */
-      return false;
+    struct stat stats;
 
-    /* ignore virtual internal ports on Mac OS X (and probably other
-       BSDs) */
-    if (*t >= 'p' && *t <= 'w')
-      return false;
+    stat(path, &stats);
 
-    /* filter out "/dev/tty0", ... (valid integer after "tty") */
-    if (IsDigitASCII(*t))
-      return false;
+    return S_ISDIR(stats.st_mode);
+}
 
+[[gnu::pure]]
+static bool
+CheckTTYName(const char *name) noexcept
+{
+  /*
+   * rfcomm devices do not have a 'device/driver' file that is used
+   * to exclude virtual/pseudo terminals
+   */
+  if (StringStartsWith(name, "rfcomm"))
     return true;
-  } else if (StringStartsWith(name, "rfcomm"))
-    return true;
-  else
+
+  /*
+   * only entries with existing 'device/driver' folder
+   * aren't virtual/pseudo terminals
```

Comment:
The enum list is capped at 128 entries. My initial fix was to set that to 256. As you can see above @lordfolken proposed a fix with tighter filtering.
If you mean _Why are we doing this?_: Probably because @lordfolken thinks scrolling through >128 entries is not user friendly.
If you mean _Why is device/driver sorting out pseudo terminals?_: I have no idea, it is what the Stackoverflow-Crowd-Knowledge spit out.
Please give me a hint in which way you think this comment should be improved.

Reply:
The comment should explain why the method implemented in the code can effectively filter out ptys, while preserving physical ttys. "Somebody on stackoverflow said..." is not a good explanation.